### PR TITLE
Initial draft for secured message API

### PR DESCRIPTION
**Pull request by steven-bellock**
Signed-off-by: Steven Bellock <sbellock@nvidia.com>

### CODE REVIEW

File: doc/api/secured_message_api.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    - The operation was successful.
+- `false`
+    - The operation was not successful.
+
+### Details
+This function should only be called after the session has been fully established and
+`libspdm_secured_message_get_session_state()` returns `LIBSPDM_SESSION_STATE_ESTABLISHED`.
+
+The structure is packed and is layed out as
+- Struct Version (32 bytes)
```
Comment (jyao1) [inline doc/api/secured_message_api.md:129]: 4 bytes, or 32 bits.

File: doc/api/secured_message_api.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+- AEAD Key Size (32 bytes)
+- AEAD IV Size (32 bytes)
+- Requester Direction Encryption Key (AEAD Key Size bytes)
+- Requester Direction Salt (AEAD IV Size bytes)
+- Requester Direction Sequence Number (64 bytes)
```
Comment (jyao1) [inline doc/api/secured_message_api.md:134]: 8 bytes, or 64 bits.

File: doc/api/secured_message_api.md (hunk 1 above)
Comment (steven-bellock) [reply doc/api/secured_message_api.md]: Thanks. It has been updated.

File: doc/api/secured_message_api.md (hunk 2 above)
Comment (steven-bellock) [reply doc/api/secured_message_api.md]: Thanks. It has been updated.

File: doc/api/secured_message_api.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+- `false`
+    - The operation was not successful.
+
+### Details
+This function should only be called after the session has been fully established and
+`libspdm_secured_message_get_session_state()` returns `LIBSPDM_SESSION_STATE_ESTABLISHED`.
+
+The structure is packed and is layed out as
+- Struct Version (4 bytes)
+- AEAD Key Size (32 bytes)
```
Comment (jyao1) [inline doc/api/secured_message_api.md:130]: this is also 4 bytes.

Comment (steven-bellock) [reply doc/api/secured_message_api.md:130]: Ok I'll fix it in the next revision.

File: doc/api/secured_message_api.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+- AEAD IV Size (32 bytes)
```
[the remark this answers is not in the document]
Comment (steven-bellock) [reply doc/api/secured_message_api.md:131]: Ok I'll fix it in the next revision.

**Pull request merged**